server/__test__/index.test.js

```diff
@@ -229,9 +232,51 @@ describe('authentication', () => {
 
     test('should get news from the news table', async () => {
       const { rows } = await getNews();
-      return expect(rows[5].content).toEqual(
+      const allContent = rows.map((row) => row.content);
+      return expect(allContent).toContain(
         'Researchers simulated a tailgating situation with a small group of overweight but healthy men and examined the impact of eating and drinking on their livers using blood tests and a liver scan.',
       );
     });
+
+    describe('health news', () => {
+      it('router returns 200 after sending the news', async () => {
+        const { status } = await request(app)
+          .get('/api/v1/healthnews/')
+          .expect(200);
+        expect(status).toBe(200);
+      });
+    });
+
+    describe('getFoodCategory query', () => {
+      test('should get Food in Category from the userFoodRelation table', async () => {
+        const { rows } = await getFoodCategory(4, 2);
+        return expect(rows[0].food_name).toEqual('maqlobah');
+      });
+    });
+
+    describe('food category', () => {
+      it('router returns 401 if there is no user logged', async () => {
+        const { status } = await request(app)
+          .get('/api/v1/user/0/category/1/food')
+          .expect(401);
+        expect(status).toBe(401);
+      });
+
+      it('router returns 200 if there is no food in the category', async () => {
+        const { status } = await request(app)
+          .get('/api/v1/user/4/category/1/food')
+          .set('Cookie', [`token=${tokenCategory}`])
+          .expect(200);
+        expect(status).toBe(200);
+      });
+
+      it('router returns 200 if there is food in the category', async () => {
+        const { status } = await request(app)
+          .get('/api/v1/user/4/category/2/food')
+          .set('Cookie', [`token=${tokenCategory}`])
+          .expect(200);
+        expect(status).toBe(200);
```

Comment:
And this test could assert that the request's `data` has the food items you expect.

Reply:
I tried doing this at first but I didn't know how, I searched for a way to do it and the logic I found was to use isArray(), then in the test use toBeTruthy() but it didn't seem to be a logical approach so I ended up not doing it.

Reply:
I think you could do expect(data).toEqual(database food data goes here, for example [{foodId: 1, name: “chicken”, .. }])